azurelinuxagent/ga/exthandlers.py

```diff
@@ -1350,13 +1350,26 @@ def get_handler_status(self):
         if not os.path.isfile(status_file):
             return None
 
+        handler_status_contents = ""
         try:
-            data = json.loads(fileutil.read_file(status_file))
+            handler_status_contents = fileutil.read_file(status_file)
+            data = json.loads(handler_status_contents)
             handler_status = ExtHandlerStatus()
             set_properties("ExtHandlerStatus", handler_status, data)
             return handler_status
         except (IOError, ValueError) as e:
             self.logger.error("Failed to get handler status: {0}", e)
+        except Exception as e:
+            error_msg = "Failed to get handler status message: {0}.\n Contents of file: {1}".format(
+                ustr(e), handler_status_contents).replace('"', '\'')
+            add_periodic(
```

Comment:
Adding a periodic event and not a custom logic for periodic as the object of `ExtHandlerInstance` class is created on the fly and cant hold state. 
(The alternative was a bit messy, check out my previous commit for a peek - https://github.com/Azure/WALinuxAgent/pull/1846/commits/e7fe92873340f49864456c3153d3ff4c58d89a30